p2panda-sync/src/log_sync.rs

```diff
@@ -263,7 +261,7 @@ where
     let log = store
         .get_raw_log(public_key, log_id)
         .await
-        .map_err(|err| SyncError::Custom(format!("could not retrieve log from store: {err}")))?;
+        .map_err(|err| SyncError::Critical(format!("could not retrieve log from store, {err}")))?;
```

Comment:
Follow up on the discussion in PR https://github.com/p2panda/p2panda/pull/569#discussion_r1810248002:

Storage errors are _not_ I/O errors, as potentially nothing went wrong with the communication between us and the database but with the database itself. On top semantically it's more interesting to know if it's an critical system error and we should maybe worry about it